test/test_cpp_api_parity.py

```diff
@@ -225,13 +252,26 @@ def _test_torch_nn_module_ctor_args(self, module_name):
         cpp_module_option = 'torch::nn::{}Options{}'.format(module_name, cpp_default_constructor_args_str)
         init_kwargs = init_arg_spec.args[-len(init_kwargs_defaults):]
         for arg_name, python_default_value in zip(init_kwargs, init_kwargs_defaults):
-            cpp_module_option += '.{}({})'.format(arg_name, self._python_arg_to_cpp_arg(python_default_value).value)
+            # NOTE: If a Python module constructor arg's default value is None, we don't test its corresponding
+            # options arg in C++ module (because the way to set the C++ options arg to an empty value is to not
+            # specify it, which means we can't test that the options arg exists).
+            # Instead, we test that all options args exist by calling their accessors after constructing the
+            # C++ module with the options.
+            if python_default_value is not None:
+                cpp_module_option += '.{}({})'.format(arg_name, self._python_arg_to_cpp_arg(python_default_value).value)
+
+        extra_stmts = []
+        for arg_name in python_default_constructor_arg_names + init_kwargs:
+            extra_stmts.append(TORCH_NN_MODULE_TEST_OPTIONS_ARG.substitute(
+                options_arg_name=arg_name,
+            ))
 
         cpp_sources = TORCH_NN_MODULE_COMMON_TEST_HARNESS + module_metadata.cpp_sources
         cpp_sources += TORCH_NN_MODULE_TEST_CTOR_ARGS.substitute(
             module_name=module_name,
             module_qualified_name='torch::nn::{}'.format(module_name),
-            module_option=cpp_module_option)
+            module_option=cpp_module_option,
+            extra_stmts=''.join(extra_stmts))
```

Comment:
Now for some reading comprehension: what happens if the C++ module has an extra option that the python module doesn't have?

Reply:
I am planning to change the internal data structure of all C++ module / optimizer options to a map, after which we will be able to check for extra C++ module options by removing all the known options from the map and see if there is anything left.